HyperPlay-Gaming/hyperplay-desktop-client
# [macOS - Feat] Add support for Apple Gaming porting toolkit

With this, Hyperplay will look for a current setup of the apple's GPTK and list it under the Wine List


## HOW TO TEST: Downloaded Games

- To download the build to test, go bellow under 'CHECKS' and build_macOS click on `details`. Then on the page that will open, click on SUMMARY. And then go to ARTIFACTS and download the build for your system.
- Since this build is NOT signed (only releases are), you need to first run this command on the terminal: `xattr -d -r com.apple.quarantine PATH_TO/HyperPlay.app`
- Make sure you have installed the gaming porting toolkit following the Install instructions provided by Apple.
- Make sure Spotlight is enabled and the command `mdfind wine64` returns something on the terminal.
- Try to run some games and make sure it is working already.
- When opening HyperPlay, go to Settings and on Game default check if, under WINE, it shows the GPTK option on the list

- Download and install some Windows games and select the Script you want to use on the game wine settings.
- Use new prefixes instead of reusing an existing one. If you have DXVK installed on the prefix, the game might not work.

## Sideloaded Games

- Tested only Steam so far
- Click on ADD GAMES on the library.
- Fill in the title as you want.
- Select the Gaming Toolkit script you want to use.
- Click on RUN INSTALLER FIRST and select the exe you want to install
- The app will be installed inside the prefix
- On Select EXE, select the exe to run the app, for instance: `/Users/name/Games/GPT_PREFIX/drive_c/Program Files (x86)/Steam/Steam.exe`
- Click Finish

---

Use the following Checklist if you have changed something on the Backend or Frontend:
